File: WorldFeed/Server/src/WorldFeed.Common/DTO/MessageEventWithAppDTO.cs
namespace WorldFeed.Common.DTO
{
    using Newtonsoft.Json;

    using WorldFeed.Common.JsonConverters;
    using WorldFeed.Common.Public.Models.Interfaces;
    using WorldFeed.Common.Public.Models.Interfaces.DTO;
    using WorldFeed.Common.Public.Models.Interfaces.DTO.Events;

    public class MessageEventWithAppDTO : IMessageEventWithAppDTO
    {
        [JsonProperty("event")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public IMessageEventDTO MessageEvent { get; set; }

        [JsonProperty("app")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public IApp App { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Core/DTO/MessageEventWithAppDTO.ts
import {IApp} from "../../Public/Models/Interfaces/IApp";
import {IMessageEventDTO} from "../../Public/Models/Interfaces/DTO/Events/IMessageEventDTO";
import {IMessageEventWithAppDTO} from "../../Public/Models/Interfaces/DTO/IMessageEventWithAppDTO";

export class MessageEventWithAppDTO implements IMessageEventWithAppDTO {
  // [JsonProperty("event")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public messageEvent: IMessageEventDTO;

  // [JsonProperty("app")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public app: IApp;
}
